### PR TITLE
Document & Dataset component

### PR DESCRIPTION
**Pull request by KhadijaMahanga**
## Description

Add Document & Dataset component 

Fixes [#172101693](https://www.pivotaltracker.com/story/show/172101693)

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
Screenshot from 2020-04-20 20-49-08
Screenshot from 2020-04-20 20-47-56

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -2,8 +2,11 @@
   "name": "@commons-ui/core",
   "version": "0.0.0",
   "dependencies": {
+    "next": "^9.3.5",
```
Comment (kilemensi) [inline packages/core/package.json]:
```suggestion
```
:exploding_head:

File: packages/core/src/Link.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
```
Comment (kilemensi) [inline packages/core/src/Link.js]: `core` needs to be reusable in any `React` project... Not necessarily Nextjs.

**Review by kilemensi — COMMENTED** [review]
:+1: @KhadijaMahanga 

1. It's broken on mobile
Screenshot_2020-04-17 Components DocumentsAndDatasets - Default ⋅ Storybook
2. Since they've gone to a lot of trouble getting these images, I think it's best to make sure the image is always visible i.e. it just grows and shrinks but doesn't disappear/get cutoff for `>= md` screen sizes:
Screenshot from 2020-04-17 17-51-19

PS: You already know it but just wanted to point out we have updated styles for this component already.

**Review by nerdlet — COMMENTED** [review]
@KhadijaMahanga  same as @kilemensi  above  :point_up_2:

File: packages/core/src/DocumentsAndDatasets/index.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        </Grid>
+        <Grid item sm={12} md={4} container className={classes.datasetData}>
+            <Content 
+                children={datasetContent.children}
+                title={datasetContent.title}
+                contentCount={datasetContent.contentCount}
+                contentType={datasetContent.contentType}
+                description={datasetContent.description}
+                linkTitle={datasetContent.linkTitle}
+                link={datasetContent.link}
```
Comment (kilemensi) [inline packages/core/src/DocumentsAndDatasets/index.js]: We need to be able to customize `Content` classes as well.. I don't see you _forwarding_ them here.

**Review by kilemensi — COMMENTED** [review]
:+1: @KhadijaMahanga 

1. Highlight image needs to have colour (blue) too:
Screenshot from 2020-04-20 08-08-15
2. Lets use `@material-ui/icons` as much as possible in COMMONS-UI.

**Comment by KhadijaMahanga** [comment]
@kilemensi reworked on your comments. With the image highlight, the image piero provided has a white background. Still waiting to here from him for one with transparent background

**Review by kilemensi — COMMENTED** [review]
Lets fix this alignment issue (Document & Dataset are not on the same line). @KhadijaMahanga, `lint` & merge. :+1: 

Screenshot_2020-04-20 Components DocumentsAndDatasets - Default ⋅ Storybook

**Pull request merged**